phygnn/model_interfaces/phygnn_model.py

```diff
@@ -0,0 +1,618 @@
+        optimizer : tensorflow.keras.optimizers, optional
+            Instantiated neural network optimization object.
+            None defaults to Adam.
+        learning_rate : float, optional
+            Optimizer learning rate.
+        history : None | pd.DataFrame, optional
+            Learning history if continuing a training session.
+        kernel_reg_rate : float, optional
+            Kernel regularization rate. Increasing this value above zero will
+            add a structural loss term to the loss function that
+            disincentivizes large hidden layer weights and should reduce
+            model complexity. Setting this to 0.0 will disable kernel
+            regularization.
+        kernel_reg_power : int, optional
+            Kernel regularization power. kernel_reg_power=1 is L1
+            regularization (lasso regression), and kernel_reg_power=2 is L2
+            regularization (ridge regression).
+        bias_reg_rate : float, optional
+            Bias regularization rate. Increasing this value above zero will
+            add a structural loss term to the loss function that
+            disincentivizes large hidden layer biases and should reduce
+            model complexity. Setting this to 0.0 will disable bias
+            regularization.
+        bias_reg_power : int, optional
+            Bias regularization power. bias_reg_power=1 is L1
+            regularization (lasso regression), and bias_reg_power=2 is L2
+            regularization (ridge regression).
+
+        Returns
+        -------
+        model : PhygnnModel
+            Initialized PhygnnModel instance
+        """
+        if isinstance(label_names, str):
+            label_names = [label_names]
+
+        model = cls.compile_model(p_fun,
+                                  loss_weights=loss_weights,
+                                  n_features=len(feature_names),
+                                  n_labels=len(label_names),
+                                  hidden_layers=hidden_layers,
+                                  metric=metric,
+                                  initializer=initializer,
+                                  optimizer=optimizer,
+                                  learning_rate=learning_rate,
+                                  history=history,
+                                  kernel_reg_rate=kernel_reg_rate,
+                                  kernel_reg_power=kernel_reg_power,
+                                  bias_reg_rate=bias_reg_rate,
+                                  bias_reg_power=bias_reg_power,
+                                  feature_names=feature_names,
+                                  output_names=label_names)
+
+        model = cls(model, feature_names=feature_names,
+                    label_names=label_names, normalize=normalize)
+
+        return model
+
+    @classmethod
+    def train(cls, p_fun, features, labels, p, normalize=(True, False),
```

Comment:
I still feel like its weird to have a class method that builds and trains and is just called train. A lot of these methods are ambiguously named. Maybe i'm being annoying by continually re-hashing this but what about having the class methods called "make_trained_model()" or something like that? Feel free to tell me this isnt worth it.

Reply:
Nomenclature is the worst name. I'm all for making them less ambiguous, that said I seem to have an aversion of overly long method names. What about just trained_model since its a class method? I can make TfModel consistent

Reply:
Looks like we've got build_trained(). I like it!